lua/intro.lua

```diff
@@ -0,0 +1,149 @@
+        {
+            "Sponsor Vim development!",
+            "type  :help sponsor<Enter>    for information ",
+        },
+        {
+            "Become a registered Vim user!",
+            "type  :help register<Enter>   for information ",
+        },
+        {
+            "Help poor children in Uganda!",
+            "type  :help iccf<Enter>       for information ",
+        },
+    }
+
+    -- Show the sponsor and register message one out of four times, the Uganda
+    -- message two out of four times.
+    local help_type = math.random(4)
+    if help_type <= 2 then
+        vim.list_extend(lines, help_messages[help_type])
+    else
+        vim.list_extend(lines, help_messages[3])
+    end
+
+    local longest_line = 0
+    for i, line in ipairs(lines) do
+        if line:len() > 0 then
+            -- translate
+            line = vim.fn.gettext(line)
+        end
+        longest_line = math.max(longest_line, line:len())
+        lines[i] = line
+    end
+
+    -- add padding to the lines
+    for i, line in ipairs(lines) do
+        local padding = math.floor((longest_line - line:len()) / 2) + 1
+        lines[i] = string.rep(" ", padding) .. line
+    end
+
+    -- add highlight to the special keys
+    vim.api.nvim_buf_set_lines(buf, 0, -1, false, lines)
+    for i, line in ipairs(lines) do
+        local s, e = line:find("<.*>")
+        if s ~= nil then
+            vim.api.nvim_buf_add_highlight(buf, -1, "SpecialKey", i - 1, s - 1, e)
+        end
+    end
+
+    local width = longest_line + 2
+    local height = #lines
+
+    local function get_config()
+        return {
+            relative = "editor",
+            width = width,
+            height = height,
+            col = math.max(0, math.ceil((vim.o.columns - width) / 2)),
+            row = math.max(0, math.ceil((vim.o.lines - height) / 2)),
+            style = "minimal",
+            border = "none"
+        }
+    end
+
+    local win = vim.api.nvim_open_win(buf, false, get_config())
+    vim.api.nvim_win_set_option(win, "winhighlight", "NormalFloat:Normal")
+
+    local win_autogroup = vim.api.nvim_create_augroup("neovide_intro_win", { clear = true })
+
+    local function close()
+        vim.api.nvim_del_augroup_by_id(win_autogroup)
+        if vim.api.nvim_buf_is_valid(buf) then
+            vim.api.nvim_buf_delete(buf, { force = true })
+        end
+        if vim.api.nvim_win_is_valid(win) then
+            vim.api.nvim_win_close(win, true)
+        end
+    end
+
+    vim.api.nvim_create_autocmd({ "VimResized" }, {
+        pattern = "*",
+        group = win_autogroup,
+        callback = function()
+            vim.api.nvim_win_set_config(win, get_config())
+        end
+    })
+
+    local first_cursor_move = false
+
+    vim.api.nvim_create_autocmd(
+        {
+            "BufEnter",
+            "ModeChanged",
+            "CursorMoved", "CursorMovedI",
+            "TextChanged", "TextChangedI", "TextChangedP", "TextChangedT"
+        },
```

Comment:
The original intro in TUI Neovim vanishes as soon as any character is typed in command mode. The intro here atm however does not, and persists even after a command has been run (excluding buffer-interacting stuff like `:normal`).

```suggestion
        {
            "BufEnter",
            "ModeChanged",
            "CursorMoved", "CursorMovedI",
            "TextChanged", "TextChangedI", "TextChangedP", "TextChangedT", "CmdlineChanged",
        },
```

Reply:
I left it like that so you don’t have to remember the full help command to type. I think it makes more sense that way.